### ISSUE TITLE
Use CocoaMQTT (1.1.3) work but I up CocoaMQTT (1.2.0) not work. Why?

### ISSUE THREAD

**Issue by chanoktrue** [emqtt/CocoaMQTT#259]
Use Code same two version
<img width="802" alt="Screen Shot 2019-05-07 at 11 08 35" src="https://user-images.githubusercontent.com/3993516/57271178-aa99a480-70b8-11e9-8ac7-9455658efbaf.png">

CocoaMQTT (1.1.3) work 
<img width="692" alt="Screen Shot 2019-05-07 at 11 06 46" src="https://user-images.githubusercontent.com/3993516/57271204-c2712880-70b8-11e9-84a3-4f82634a8244.png">

CocoaMQTT (1.2.0) not work
<img width="504" alt="Screen Shot 2019-05-07 at 11 01 30" src="https://user-images.githubusercontent.com/3993516/57270980-be90d680-70b7-11e9-88a7-f5f713ba40b2.png">

**Comment by HJianBo** [comment]
Hi, @chanoktrue thanks for your feedback this. I'll test it later!

**Comment by chanoktrue** [comment]
Thx.

**Comment by captainOfM** [comment]
jap, same and for me

**Comment by keshavx11** [comment]
+1

**Comment by wickyou23** [comment]
same for me :(

**Comment by KevinCoderX** [comment]
```
mqtt.subscribe(subTopicPrefix + "+", qos: .qos0)
```

It will disconnect

**Comment by keshavx11** [comment]
> ```
> mqtt.subscribe(subTopicPrefix + "+", qos: .qos0)
> ```
> 
> 
> It will disconnect

For me, I'm trying to subscribe on

```
mqtt.subscribe(subTopicPrefix + "#", qos: .qos1)
```

So possible that issue is due to special characters in topic

**Comment by HJianBo** [comment]
Hi, @chanoktrue I have followed your code, but it works well for me.  Can you send me an example project to heeejianbo@gmail.com?

**Comment by HJianBo** [comment]
@KevinCoderX Yes, maybe you should check the Topic format is allowable.

These topics is illegal:
```
a/b+
a/b#
a/#/b
```

The correct format is:
```
a/b/+
a/+/b
a/b/#
```

**Comment by KevinCoderX** [comment]
My topic is "dadsafafa/+", My team need it.

**Comment by wickyou23** [comment]
Hi @HJianBo, 

I checked my code and found an issue that after subscribing to topic "a/b/#",  the socket lost connect immediately, but if I remove that line code, it works well. At the version 1.1.3, it works well.

**Comment by TiSiAi** [comment]
Same problem for me.
When I subscribe, it disconnect. 
It's OK with 1.1.3

**Comment by HJianBo** [comment]
Guys, Which Broker did you use for testing? I have tested it with EMQ X Broker. it works well.

**Comment by ioschen** [comment]
same and for me

**Comment by TiSiAi** [comment]
> Guys, Which Broker did you use for testing? I have tested it with EMQ X Broker. it works well.

For me Mosquitto ( windows last version )
Thank !

**Comment by keshavx11** [comment]
Verne MQ for me

**Comment by bellots** [comment]
same problem for me. Downgraded to 1.1.3 works!

**Comment by xavierzhao** [comment]
pod 'CocoaMQTT', '~> 1.1.3'

**Comment by DpoH** [comment]
Hi guys. I've just created [PR](https://github.com/emqtt/CocoaMQTT/pull/261) fixes this issue. Please check it out.

**Comment by TiSiAi** [comment]
It's seem OK for me with Mosquitto. 
But with an update CocoaPod or clean install with 1.2.0 or last  it's fail, I must after an update or clean install copy manually the content of the fix and replace all CocoaMQTTFrame.swift content file.
I test next day in long
Thank !

**Comment by DpoH** [comment]
You could use follow pod-instruction
pod 'CocoaMQTT', :git => 'https://github.com/emqtt/CocoaMQTT.git'
till pod version will be released.

**Comment by ioschen** [comment]
1.2.1 still has problems, I am testing multiple times

**Comment by HJianBo** [comment]
Hi, @ioschen Can you provide the Broker name, version and paste the subscribe code here?

**Comment by ioschen** [comment]
> Hi, @ioschen Can you provide the Broker name, version and paste the subscribe code here?

1.2.1 No big problem, just disconnected twice, I have changed to 1.1.3
If there is a disconnection in 1.2.1, I will give feedback again. 
I will see if it is my own problem.

**Comment by HJianBo** [comment]
Hi guys, This problem has fixed at v1.1.3. Just close it
